crates/bevy_render/src/extract_param.rs

```diff
@@ -52,8 +52,10 @@ pub struct ExtractState<P: SystemParam + 'static> {
     main_world_state: <Res<'static, MainWorld> as SystemParam>::State,
 }
 
-// SAFETY: only accesses MainWorld resource with read only system params using Res,
-// which is initialized in init()
+// SAFETY: The only `World` access is read-only.
+unsafe impl<P> ReadOnlySystemParam for Extract<'_, '_, P> where P: ReadOnlySystemParam {}
```

Comment:
We should also make sure that `Res<'w, MainWorld>` is `ReadOnlySystemParam`.

Reply:
`Res<>` is always `ReadOnlySystemParam`, and `Res<MainWorld>` is the only `World` access used here. I added a note to clarify that I was referring to `Res<MainWorld>`.